tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -232,181 +309,170 @@ def _setup(self) -> None:
+                    self._lisa_log.info("Downloading %s to %s", pypy_download, pypy)
+                    log.info("Downloading %s to %s", pypy_download, pypy)
+                    run_command(["wget", pypy_download, "-O",  pypy])
+
+            #
+            # Create a tarball with the tools we need to copy to the test node. The tarball includes two directories:
+            #
+            #     * bin - Executables file (Bash and Python scripts)
+            #     * lib - Library files (Python modules)
+            #
+            self._lisa_log.info("Creating %s with the tools needed on the test node", self._test_tools_tarball_path)
+            log.info("Creating %s with the tools needed on the test node", self._test_tools_tarball_path)
+            log.info("Adding orchestrator/scripts")
+            command = "cd {0} ; tar cf {1} --transform='s,^,bin/,' *".format(self._test_source_directory/"orchestrator"/"scripts", self._test_tools_tarball_path)
+            log.info("%s", command)
+            run_command(command, shell=True)
+            log.info("Adding tests/scripts")
+            command = "cd {0} ; tar rf {1} --transform='s,^,bin/,' *".format(self._test_source_directory/"tests"/"scripts", self._test_tools_tarball_path)
+            log.info("%s", command)
+            run_command(command, shell=True)
+            log.info("Adding tests/lib")
+            command = "cd {0} ; tar rf {1} --transform='s,^,lib/,' --exclude=__pycache__ tests_e2e/tests/lib".format(self._test_source_directory.parent, self._test_tools_tarball_path)
+            log.info("%s", command)
+            run_command(command, shell=True)
+            log.info("Contents of %s:\n%s", self._test_tools_tarball_path, run_command(['tar', 'tvf', str(self._test_tools_tarball_path)]))
 
             log.info("Completed setup, creating %s", completed)
             completed.touch()
 
         finally:
             self._setup_lock.release()
 
-    def _build_agent_package(self) -> None:
-        """
-        Builds the agent package and returns the path to the package.
-        """
-        log.info("Building agent package to %s", self.context.working_directory)
-
-        makepkg.run(agent_family="Test", output_directory=str(self.context.working_directory), log=log)
-
-        package_path: Path = self._get_agent_package_path()
-        if not package_path.exists():
-            raise Exception(f"Can't find the agent package at {package_path}")
-
-        log.info("Built agent package as %s", package_path)
-
-    def _get_agent_package_path(self) -> Path:
+    def _clean_up(self, success: bool) -> None:
         """
-        Returns the path to the agent package.
+        Cleans up any items created by the test suite run.
         """
-        return self.context.working_directory/"eggs"/f"WALinuxAgent-{AGENT_VERSION}.zip"
+        if self._delete_scale_set:
+            if self._keep_environment == KeepEnvironment.Always:
```

Comment:
If I remember correctly, I think lisa always skips clean up if we provide an existing vm (even if keep_environment is not always). Do we want to keep that behavior here?

Reply:
That is already taken care of by the check on self._delete_scale_set, which is set to True only when the AgentTestSuite itself creates the VM.

LISA does log an additional message about not deleting existing VMs. It seems redundant to me, but if you think it is useful I can add a similar message here, though that would require keeping an extra flag.

Reply:
I don't think the log is necessary either